LoRaEngine/modules/LoRaWanNetworkSrvModule/Logger/TcpLogger.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace Logger
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Net;
+    using System.Net.Sockets;
+    using LoRaWan;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.DependencyInjection.Extensions;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Logging.Configuration;
+
+    internal sealed class TcpLoggerProvider : ILoggerProvider
```

Comment:
This type is only used in `TcpLoggerExtensions`, can we make it nested there and also make it private to limit its visibility?

Reply:
I tried to be consistent with the `LoRaConsoleLoggerProvider`. It doesn't seem right to me to hide a class that is required for custom loggers within extensions - open to discuss this if we should pick it up again.

Reply:
Addressed with f8ebe5e28f671cebf0f3f306a48ebc7d474391fb